app/domain/authentication/authn_k8s/resource_restrictions.rb

```diff
@@ -3,140 +3,89 @@
 module Authentication
   module AuthnK8s
 
-    # This class defines an application identity of a given conjur host.
-    # The constructor initializes an ApplicationIdentity object and validates that
-    # it is configured correctly.
-    # The difference between the validation in this constructor and
-    # the validation in ValidateApplicationIdentity is that here we validate that
-    # the application identity is configured correctly, and thus is a valid application
-    # identity. In ValidateApplicationIdentity we validate that the defined application
-    # identity is actually the correct one in kubernetes.
-    # For example, an application identity `some-namepsace/blah/some-value` is not
-    # valid and will fail the validation here. However, an application identity
-    # `some-namespace/service-account/some-value` is a valid application identity
-    # and will pass the validation here. If the host is actually running from
-    # a pod with service account `some-other-value` then it will fail the
-    # validation of ValidateApplicationIdentity
-    class ApplicationIdentity
-
-      AUTHENTICATION_CONTAINER_NAME_ANNOTATION = "authentication-container-name"
-      DEFAULT_AUTHENTICATION_CONTAINER_NAME = "authenticator"
-
-      def initialize(host_id:, host_annotations:, service_id:)
+    # This class represents the restrictions that are set on a Conjur host regarding
+    # the K8s resources that it can authenticate with Conjur from.
+    # It consists a list of K8sResource objects which represent the resource
+    # restriction that need to be met in an authentication request.
+    #
+    # For example, if `resources` includes the K8sResource:
+    #   - type: "namespace"
+    #   - value: "some-namespace"
+    #
+    # then this Conjur host can authenticate with Conjur only from a pod that is
```

Comment:
I would mention here for clarity purposes that in authn-k8s there are two ways that resource restrictions can be provided for authentication 1. via host annotations (newer) 2. via host id (older)